### PR TITLE
Recipe App Version 1.0

### PR DESCRIPTION
**Pull request by asharanjith**
Hi we are a team of two @AhmadiGu @asharanjith 

In this PR we created the Food-Recipes App with the following features and based on the [ER diagram](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/recipe_erd_2_members.png)

- [x] Login page and registration page:
-  Should be built with Devise.
- [x] Food list:
- Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
- Should lead to a form that allows users to add new food.
- [x] Recipes list:
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.
- [x] Public recipe list:
- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
-  [x] Recipe details:
- Should display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
-  [x] General shopping list view 
- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.
- [x] Create a navigation menu that allows users to open all of the pages you created.
- [x] Write unit and integration tests

### CODE REVIEW

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class FoodsController < ApplicationController
+  def index
+    @user = current_user
+    @foods = Food.where(user: current_user) || []
+  end
```
Comment (mckent05) [inline app/controllers/foods_controller.rb:5]: - You have done well so far, but according to the project requirements, a logged in user should be able to see all available food irrespective of the owner of the food. Please fix this 🙏. And remember only the owner of a food should be able to delete it.

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @user = current_user
+    @recipes = @user.recipes.includes(:user)
```
Comment (mckent05) [inline app/controllers/recipes_controller.rb:7]: - Just like for foods, a logged in user should be able to see all recipes irrespective of the owner of the recipe, please fix this 🙏. Also remember that only the owner of a recipe can delete the recipe.

File: app/controllers/foods_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  def new
+    @food = Food.new
+    render 'new'
+  end
+
+  def create
+    @food = Food.new(food_params)
+    @food.user = current_user
```
Comment (mckent05) [inline app/controllers/foods_controller.rb:14]: **OPTIONAL** - To help make your code cleaner and more coincise, you could do `current_user.foods.new(food_params`

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi guys, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Your project looks good and almost ready 👏 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmetbozaci** [comment]
### Hello @AhmadiGu @AhmadiGu 
### _This is an additional review as a result of the second opinion request._ 
- As per the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=Should%20display%20a%20list%20of%20food,in%20user%20as%20in%20the%20wireframe.), the logged-in user should be able to view a list of food and recipe items. However, not all of these items are created by other users. I am pleased to inform you that you have already implemented this requirement.




___
**_As this was the only reason your project was not approved it will be approved as soon as possible._**

_**Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away**_


### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Asha & Gulalai,  

Your project is complete! 🏁 Great job 💎 💎 💎! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉 💯 🌟 

## Highlights 🎯 
💠  Recipe all is working as intended!
💠 Professional ReadMe!

 keep soaring higher! 💻  🍷  🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**